packages/inferno-router/src/resolveLoaders.ts

```diff
@@ -0,0 +1,198 @@
+import { isNullOrUndef, isUndefined } from "inferno-shared";
+import { matchPath } from "./matchPath";
+import type { TLoaderData, TLoaderProps } from "./Router";
+import { Switch } from "./Switch";
+
+export function resolveLoaders(loaderEntries: TLoaderEntry[]): Promise<Record<string, TLoaderData>> {
+  const promises = loaderEntries.map((({ path, params, request, loader }) => {
+    return resolveEntry(path, params, request, loader);
+  }));
+  return Promise.all(promises).then((result) => {
+    return Object.fromEntries(result);
+  });
+}
+
+type TLoaderEntry = {
+  path: string,
+  params: Record<string, any>,
+  request: Request,
+  controller: AbortController,
+  loader: (TLoaderProps) => Promise<TLoaderEntry>,
+}
+
+// Optionally pass base param during SSR to get fully qualified request URI passed to loader in request param
+export function traverseLoaders(location: string, tree: any, base?: string, parentIsSwitch = false): TLoaderEntry[] {
```

Comment:
hmm, so this method is exposed to public and contains a boolean information about parent node being a switch? `parentIsSwitch` I would like to get that flag removed if possible

Reply:
This method could be written using a stack of nodes to avoid recursion and avoid exposing switch parameter externally

Reply:
a minimum change could be to have a separate method for traverseLoaders which has parentIsSwitch and call that internally here and dont expose it

Reply:
WILL FIX: I'll go for exposing a public method

Reply:
Fixed in 0843db8fcb